src/Promitor.Agents.Scraper/Program.cs

```diff
@@ -20,16 +22,26 @@ public static int Main(string[] args)
                 // Let's hook in a logger for start-up purposes.
                 ConfigureStartupLogging();
 
-                CreateHostBuilder(args)
-                    .Build()
-                    .Run();
+                var host = CreateHostBuilder(args)
+                    .Build();
+                
+                using (var scope = host.Services.CreateScope())
+                {
+                    var validator = scope.ServiceProvider.GetRequiredService<RuntimeValidator>();
+                    if (!validator.Run())
+                    {
+                        return (int)ExitStatus.ValidationFailed;
+                    }
+                }
 
-                return 0;
+                host.Run();
+
+                return (int)ExitStatus.Success;
             }
             catch (Exception exception)
             {
                 Log.Fatal(exception, "Host terminated unexpectedly");
```

Comment:
@tomkerkhove what do you think about tweaking this error message? I'm not sure if you've used Terraform before, but if it encounters some kind of unexpected error (vs a config mistake that the user has made), it explicitly says "this is a bug in Terraform". I think it's kinda neat because it gives you reassurance that you (as a user) haven't done anything wrong.

Just wondering if we change it to something like "Promitor has encountered an unexpected error. This is a bug - please open a GitHub issue at https://github.com/tomkerkhove/promitor/issues"?

Reply:
I'm happy to change it but "this is a bug in Promitor" as we're not sure so we should add a "possible" 😄 

I like your suggestion at the end!